src/main/java/io/leonis/frosty/ControllerTransformer.java

```diff
@@ -0,0 +1,57 @@
+package io.leonis.frosty;
+
+import io.reactivex.Flowable;
+import io.reactivex.functions.Function;
+import java.util.*;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import org.reactivestreams.Publisher;
+
+/**
+ * The Class ControllerTransformer.
+ *
+ * Transforms a {@link Set} of {@link Controller Controllers} to {@code O} to a {@link Map} grouped
+ * by it's applicable identity.
+ *
+ * @param <C> Type of controller state
+ * @param <A> Type of assignee identity
+ * @param <O> Type of output
+ * @author Jeroen de Jong
+ */
+@AllArgsConstructor
+public final class ControllerTransformer<
+    T extends Controller.SetSupplier<C> & Controller.MapSupplier<A>,
+    C extends Controller, A, O>
+    implements Function<Flowable<T>, Publisher<Map<A, O>>> {
+  /**
+   * Function that maps controller to the desired output type.
+   */
+  private final java.util.function.Function<C, O> outputFunction;
+
+  /**
+   * Function which reduces the accumulated set of O to O.
+   */
+  private final java.util.function.Function<Set<O>, O> outputReducer;
+
+  /**
+   * {@inheritDoc}
+   *
+   * todo this needs to be rewritten to use rxJava instead of streams
```

Comment:
i see flowable is used. do you mean to rewrite the `.map` methods?

Reply:
Yes, that way backpressure works better i guess and we can possibly compile down to a lower version of java (without the streams lib)